zebrad/tests/acceptance.rs

```diff
@@ -1,30 +1,191 @@
+    cmd.wait().unwrap().expect_code(1);
+
+    let mut runner = RUNNER.clone();
+    let cmd = runner
+        .args(&["generate", "-o", "file"])
+        .capture_stdout()
+        .run();
+    cmd.wait().unwrap().expect_success();
+
+    // Todo: Check if the file was created
+}
+
+#[test]
+fn help_no_args() {
+    let mut runner = RUNNER.clone();
+    let cmd = runner.arg("help").capture_stdout().run();
+    cmd.wait().unwrap().expect_success();
+}
+
+#[test]
+fn help_args() {
+    // Invalid argument
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["help", "argument"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+
+    // Invalid flag
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["help", "-f"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+}
+
+#[test]
+fn revhex_no_args() {
+    let mut runner = RUNNER.clone();
+    let _cmd = runner
+        .timeout(Duration::from_secs(1))
+        .arg("revhex")
+        .capture_stdout()
+        .run();
+
+    // Program is waiting for input, we just exit after 1 second
+}
+
+#[test]
+fn revhex_args() {
+    let mut runner = RUNNER.clone();
+    let mut cmd = runner.args(&["revhex", "33eeff55"]).capture_stdout().run();
+    cmd.stdout().expect_line("55ffee33");
+}
+
+#[test]
+fn seed_no_args() {
+    let mut runner = RUNNER.clone();
+    runner.exclusive();
+    let mut cmd = runner
+        .timeout(Duration::from_secs(1))
+        .arg("seed")
+        .capture_stdout()
+        .run();
+
+    // Todo: maybe add special info!() to seed command
+    // Todo: improve the regex
+    cmd.stdout()
+        .expect_regex(r"^(.*?)Initializing tracing endpoint");
+
+    // Problem: Zombie child process is not killed
+    //cmd.kill();
+}
+
+#[test]
+fn seed_args() {
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["seed", "argument"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["seed", "-f"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["seed", "start"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+}
+
+#[test]
+fn start_no_args() {
+    let mut runner = RUNNER.clone();
+    runner.exclusive();
+    let mut cmd = runner
+        .timeout(Duration::from_secs(1))
+        .arg("start")
+        .capture_stdout()
+        .run();
+
+    // Todo: maybe add special info!() to start command
+    // Todo: improve the regex
+    cmd.stdout()
+        .expect_regex(r"^(.*?)Initializing tracing endpoint");
+
+    // Problem: Zombie child process is not killed
+}
+
+#[test]
+fn start_args() {
+    // Bug? This should fail but not happening
+    //let mut runner = RUNNER.clone();
+    //let cmd = runner.args(&["start", "argument"]).capture_stdout().run();
+    //cmd.wait().unwrap().expect_code(1);
```

Comment:
Question: What does the usage documentation for `zebrad start` say about its arguments?

Does it say anything about extra arguments?
Do we need to reject extra arguments for all our commands?

Reply:
Any extra argument gets rejected automatically except if there is an actual command with that name. In the case of `start` we know it it can work combined with `seed`. All others free arguments or flags should be rejected. I will research further and see if i can make this happen.

Reply:
I got confused as the issue says in one of the items: `Seed (might be combined with Start)`

This is not actually true, the `start` command accept any free additional argument however doing `zebrad start seed` will do nothing:

```
$ ./zebrad start seed
Jul 22 14:58:47.430  INFO zebrad::components::tracing: Initializing tracing endpoint
Jul 22 14:58:47.430  INFO zebrad::components::metrics: Initializing metrics endpoint
^C
$ 
```
This extra arguments are for overwriting configuration https://github.com/ZcashFoundation/zebra/blob/main/zebrad/src/commands/start.rs#L92 but it haves nothing to do with running 2 commands in one.

I fixed the `start_args` test case accordingly in https://github.com/ZcashFoundation/zebra/pull/690/commits/9d97cfb07052a44e0351904e7db87b9a0a6cef27